lib/api/LogManagerImpl.cpp

```diff
@@ -804,15 +820,37 @@ namespace MAT_NS_BEGIN
         return m_dataViewerCollection;
     }
 
-    void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
+    void LogManagerImpl::AddOrUpdateInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            m_dataInspectors.clear();
+            return;
+        }
```

Comment:
This seems weird. Someone passes a nullptr and just clears everything?

The original design made sense for a single inspector, but I don't believe this makes sense for clearing all anymore. We probably should have a separate `ClearDataInspectors` or some method like that.

Reply:
This was just taken from the documentation: "Note: A `nullptr` may be passed to `SetInspector` if the intent is to remove any registered data inspectors."

Adding a ClearDataInspectors makes more sense and having addorupdate just do nothing when a nullptr is passed in